src/aot.s

```diff
@@ -0,0 +1,69 @@
+.text
+
+
+.global _aot_trampoline
+_aot_trampoline:
+    // x0 <- fn_ptr: extern "C" fn()
+    // x1 <- args_ptr: *const u8
+    // x2 <- args_len: usize
+
+    //
+    // Check zero arguments.
+    //
+    cmp     x2,     0                       // If there are no arguments,
+    beq     3f                              // skip argument processing completely.
+
+    //
+    // Preparation.
+    //
+    add     x1,     x1,     x2,     lsl 3   // Move the pointer to the end (past last element).
+
+    str     x19,    [sp, #-8]!              // Push the x19 register (we'll use it to store sp).
+    mov     x19,    sp                      // Store the original stack pointer for later.
+
+    //
+    // Copy stack arguments (for n_args > 8).
+    //
+    cmp     x2,     8                       // Check if there are more than 8 arguments.
+    ble     2f                              // If there are less than 8, skip to register arguments.
```

Comment:
But this would be incorrect when one argument <8 is an enum, right? Or do we report them padded?

Reply:
Yes, assuming we passed the arguments as-is to the trampoline. The idea was to prepare them in Rust and send them to the trampoline to move them where they need to be. That's why it's an array of `u64` instead of `JitValue`.